cli/src/main/java/org/jboss/sbomer/cli/feature/sbom/command/auto/GenerateEnvConfigCommand.java

```diff
@@ -0,0 +1,147 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.sbomer.cli.feature.sbom.command.auto;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import org.jboss.pnc.dto.Build;
+import org.jboss.sbomer.cli.feature.sbom.ConfigReader;
+import org.jboss.sbomer.cli.feature.sbom.command.PathConverter;
+import org.jboss.sbomer.cli.feature.sbom.service.PncService;
+import org.jboss.sbomer.core.features.sbom.enums.GenerationResult;
+import org.jboss.sbomer.core.features.sbom.utils.MDCUtils;
+import org.jboss.sbomer.core.features.sbom.utils.EnvironmentAttributesUtils;
+
+import jakarta.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
+/**
+ * <p>
+ * Command to generate a runtime environment configuration file (compliant with SDKMan) for SBOMer generation
+ * </p>
+ */
+@Slf4j
+@Command(
+        mixinStandardHelpOptions = true,
+        name = "generate-env-config",
+        description = "Generates the runtime environment configuration used for automation for a given PNC build")
+public class GenerateEnvConfigCommand implements Callable<Integer> {
+
+    public static enum ConfigFormat {
+        yaml, json;
+    }
+
+    @Option(names = { "--build-id" }, required = true, description = "The PNC build identifier, example: AYHJRDPEUMYAC")
+    String buildId;
+
+    @Option(
+            names = { "--format" },
+            defaultValue = "yaml",
+            description = "Format of the generated environment configuration.")
+    ConfigFormat format;
+
+    @Option(
+            names = { "--target" },
+            paramLabel = "FILE",
+            description = "Location where the configuration environment file should be stored. If not provided, environment configuration will be printed to standard output.",
+            converter = PathConverter.class)
+    Path target;
+
+    @Inject
+    PncService pncService;
+
+    @Inject
+    ConfigReader configReader;
+
+    /**
+     * Retrieves environment configuration from a PNC Build, compliant with SDKMan.
+     *
+     * @return a Map object if the configuration could be retrieved or {@code null} otherwise.
+     */
+    private Map<String, String> environmentConfig() {
+        // Make sure there is no context
+        MDCUtils.removeContext();
+        MDCUtils.addBuildContext(buildId);
+        log.debug("Attempting to fetch environment configuration from a PNC build");
+
+        Build build = pncService.getBuild(buildId);
+
+        if (build == null) {
+            log.warn("Could not retrieve PNC build '{}'", buildId);
+            return Collections.emptyMap();
```

Comment:
See comment below, I would check for the build earlier and probably pass it to this method.